Sources/MapboxNavigation/NavigationMapView+VanishingRouteLine.swift

```diff
@@ -238,15 +232,20 @@ extension NavigationMapView {
                     return nil
                 }
             }.reduce(0, +)
+            // lastRecordSegment records the last segmentEndPercentTraveled and associated congestion color added to the gradientStops.
+            var lastRecordSegment: (Double, UIColor) = (0.0, traversedRouteColor)
```

Comment:
`lastRecordSegment` is used to record the last congestion color added to the `gradientStops`. If the current congestion color is the same as previous one, they will be combined.